Library/Homebrew/migrated.rb

```diff
@@ -0,0 +1,19 @@
+require "tap"
+
+module Homebrew
+  # name should not be qualified, since migration of qualified names is already
+  # handled in Formulary::TapLoader.formula_name_path.
+  def search_for_migrated_formula(name, print_messages = true)
+    migrations = []
+    Tap.each do |old_tap|
+      new_tap_name = old_tap.tap_migrations[name]
+      next unless new_tap_name
+      migrations << [old_tap, new_tap_name]
+      if print_messages
+        deprecation = (new_tap_name == "homebrew/boneyard") ? "deprecated " : ""
+        puts "A #{deprecation}formula named \"#{name}\" has been migrated from #{old_tap} to #{new_tap_name}."
```

Comment:
Probably want to use `EOS.undent` and a few lines here. Test on a 80 character wide terminal and see how it looks.